services/121-service/src/registration/services/registrations-pagination.service.ts

```diff
@@ -215,27 +216,27 @@ export class RegistrationsPaginationService {
     programId,
     referenceIds,
     select,
-    chunkSize = 20000,
   }: {
     programId: number;
     referenceIds: string[];
     select?: string[];
-    chunkSize?: number;
   }): Promise<MappedPaginatedRegistrationDto[]> {
+    const chunkSize = 20000;
+
     // Ensure that the a new qb is created for a chunk of referenceIds because limited query length
     const allResults: MappedPaginatedRegistrationDto[] = [];
 
-    for (let i = 0; i < referenceIds.length; i += chunkSize) {
-      const chunk = referenceIds.slice(i, i + chunkSize);
-      const qb =
+    const chunks = chunk(referenceIds, chunkSize);
+    for (const c of chunks) {
```

Comment:
I'm good with `c` in [certain contexts](https://en.wikipedia.org/wiki/Speed_of_light) but not this one, maybe use `currentChunk`?
You can also one-line it then:
```typescript
for (const currentChunk of chunk(referenceIds, chunkSize)) {
  // ...
}
```